README.md

```diff
@@ -1,106 +1,43 @@
-
-### Basic Requirements
-
-  - Create each endpoint as an individual AWS Lambda in Node.js
-  - Use any AWS Database-as-a-Service persistence store. DynamoDB is preferred.
-  - Write concise and clear commit messages.
-  - Write clear **documentation** on how it has been designed and how to run the code.
-
-### Bonus
-
-  - Use Infrastructure-as-code tooling that can be used to deploy all resources to an AWS account. Examples: CloudFormation / SAM, Terraform, Serverless Framework, etc.
-  - Provide proper unit tests.
-  - Use API Gateway to expose AWS Lambdas
-  - Providing an online demo is welcomed, but not required.
-  - Bundle npm modules into your Lambdas
+## Written Answers
 
-### Advanced Requirements
+### Lambda Error Handling, Retries and DLQs
 
-These may be used for further challenges. You can freely skip these; feel free to try out if you feel up to it.
+When Lambda is triggered synchronously via API Gateway, there isn't AFAIK an automatic way to have the error recorded into a DLQ.  This puts the responsibility of retries on the front end application that is interacting with the API Gateway endpoint.  The approach I took in these APIs was to give clear and concise HTTP error codes to indicate whether or not a request should/could be retried.  If we were invoking Lambda in an asynchronous fashion, it would have a lot more options for retry configuration like a DLQ in SQS.
 
-  - Describe your strategy for Lambda error handling, retries, and DLQs
-  - Describe your cloud-native logging, monitoring, and alarming strategy across all endpoints
-  - Build a filter feature that can filter records with some of the attributes such as username.
+I hope this is a satisfactory answer, if we were talking about a workflow that is kicked off asynchronously there would be more to discuss about this strategy.
 
-## Q&A
+### Logging, Monitoring and alarming
 
-> Where should I send back the result when I'm done?
+Throughout the API I have made use of the console for logging, using info/error/warn where appropriate.  These logs will naturally flow into CloudWatch with the corresponding log group for each API and stage it is in (prod/dev).  Using metric filters we can pattern match and emit a value that would be alarmable with a CloudWatch alarm.  I would use a low sev/high sev pair, where a low sev may not require action but is noted that there is a potential issue and a high sev indicating that action is required immediately.  All cloud watch alarms can be modeled in IaC in the serverless.yml resource files.  This keeps any manual setup to a minimum for a developer.
 
-Fork this repo and send us a pull request when you think you are done. There is no deadline for this task unless otherwise noted to you directly.
+Since we are using managed services, each service like Lambda/API Gateway/DynamoDB already emit their own sets of metrics that can be alarmed on.  This requires fare less augmentation of the code to emit your own metrics.
```

Comment:
@GrahamEvans31 How would you recommend creating dashboards / alarms based on these aws-managed service metrics?

For example, if the team wanted to create a CloudWatch dashboard for this service via infrastructure-as-code, how could that be done via this repo?

Reply:
Haven't personally done this before, but some light reading shows that CloudFormation supports dashboarding (in the console), albeit in a painful manner with specifying the exact grid coordinates in pixels...  Some quick research shows that there are plugins available for Serverless to specify dashboard under the custom section in serverless.yml.  I'd want to explore a few of these and settle on one that allows us to describe these in our IaC and avoid any manual dashboarding.  Internal amazon was manual dashboarding, these dashboards frequently became out of touch with the underlying metrics, if metrics are added or changed, this is not a great situation to be in.